### PR TITLE
Flashloan detection bot

### PR DESCRIPTION
**Pull request by dikel**

### CODE REVIEW

File: flashloan-detector/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+- FLASHLOAN-ATTACK
+  - Fired when a transaction contains a flashoan and the borrower made significant profit
+  - Severity is always set to "high"
+  - Type is always set to "exploit"
+  - Metadata:
+    - tokens - array of all tokens involved in the transaction
+
+## Test Data
+
+The agent behaviour can be verified with the following transactions:
```
Comment (notyouraveragedev) [inline flashloan-detector/README.md]: Change agent to bot

File: flashloan-detector/src/agent.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        severity: FindingSeverity.High,
+        type: FindingType.Exploit,
+        metadata: {
+          tokens: tokensArray,
+        },
+      }));
+    }
+
+    // Clear all cached prices and delete token decimals if the object is too large
+    helper.clear();
```
Comment (notyouraveragedev) [inline flashloan-detector/src/agent.js:94]: Unt tests fails here "helper.clear() is not a function"

**Review by notyouraveragedev — CHANGES_REQUESTED** [review]
Code looks good, only two things I noticed are the things that I specified in the review

**Comment by notyouraveragedev** [comment]
Also might be a good idea explaining some of the logic behind the profit calculation with comments

**Review by notyouraveragedev — APPROVED** [review]

**Comment by notyouraveragedev** [comment]
LGTM!

File: flashloan-detector/src/flashloan-detector.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+const { getAaveFlashloan } = require('./detectors/aave-detector');
+const { getDydxFlashloan } = require('./detectors/dydx-detector');
```
Comment (christian-forta) [inline flashloan-detector/src/flashloan-detector.js:2]: are those the common ones for other chains as well?

**Review by christian-forta — CHANGES_REQUESTED** [review]
primarily I would ask to double check whether the 5 flash loan providers you identified are consistent across all the chains this bot will monitor.

File: flashloan-detector/src/flashloan-detector.js (hunk 3 above)
Comment (dikel) [reply flashloan-detector/src/flashloan-detector.js:2]: The flashloan providers emit the same events on all chains. Also the detections are not based on a particular address so they should work with all of these protocols on all chains and all of their forks

Comment (christian-forta) [reply flashloan-detector/src/flashloan-detector.js:2]: Thanks for confirming. Lets get it shipped :)
Pls send me the bot id once you do. We can keep an eye on it over the weekend and then add it to the kit early next week.

**Comment by christian-forta** [comment]
Woooo! Super excited about this bot!

**Review by stoqnkpL — APPROVED** [review]
Looks great! It will be very interesting to monitor this bot for alerts!

**Pull request merged**